compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -885,9 +885,17 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                     return;
                 }
 
-                if !tcx.is_const_fn_raw(callee) {
-                    self.check_op(ops::FnCallNonConst);
-                    return;
+                let caller_has_attr = tcx.has_attr(caller, sym::default_method_body_is_const);
+                let in_same_trait = match (tcx.trait_of_item(caller), tcx.trait_of_item(callee)) {
```

Comment:
it may potentially be desirable to only call those queries if the attribute is set. Otherwise we run these queries, even if the information is irrelevant. This can cause problems with  performance or incremental-compilation